### ISSUE TITLE
Order in no-mixed-requires

### ISSUE THREAD

**Issue by goloroden** [eslint/eslint#450]
I absolutely <3 the idea of the `no-mixed-requires` rule, especially with the `grouping` option.

What I would even more like is if there was an option to check for alphabetical order and a specific order for groups.

What I would like to have is an extension to this rule that checks the following order:
- First there is `require` for internal Node.js modules, such as `fs` or `http`
- Then there is `require` for npm-installed modules, such as `express` or `connect``
- Then there is `require` for local files, i.e. all modules that start with `./` or `../`
- Last but not least there are the normal `var` calls.

And, inside of each group, the rule should check whether the modules are alphabetically ordered (by their variable name, not by their module name).

Hence

``` javascript
var connect = require('connect'),
    express = require('express');
```

would be valid, while

``` javascript
var express = require('express'),
    connect = require('connect');
```

was not.

``` javascript
var readable = require('stream').Readable,
    compress = require('zlib');
```

again would be wrong (variables not ordered alphabetically), while

``` javascript
var compress = require('zlib'),
    readable = require('stream').Readable;
```

would be right.
